store/v2/commitment/metadata.go

```diff
@@ -158,8 +158,9 @@ func (m *MetadataStore) deleteRemovedStoreKeys(version uint64, removeStore func(
 
 	batch := m.kv.NewBatch()
 	defer func() {
-		if berr := batch.Close(); berr != nil {
-			err = berr
+		cErr := batch.Close()
```

Comment:
these are functionally identical, why the change?

Reply:
If `batch.Write()` returns error and `batch.Close()` returns error too, the error returned by `batch.Write` would be covered.

This change just keeps consistant with others: The error returned by `batch.Close` should be returned when original err is nil

Reply:
We should errors.Join them then to not lose any error.

Reply:
@julienrbrt Hi sir, I have changed to use `errors.Join` now, thanks for your review